dadi/lib/datasource/index.js

```diff
@@ -23,21 +24,90 @@ const Datasource = function (page, datasource, options) {
 
 Datasource.prototype.init = function (callback) {
   this.loadDatasource((err, schema) => {
-    if (err) {
-      return callback(err)
-    }
+    if (err) return callback(err)
 
     this.schema = schema
-    this.source = schema.datasource.source
     this.schema.datasource.filter = this.schema.datasource.filter || {}
     this.originalFilter = Array.isArray(this.schema.datasource.filter)
       ? Array.from(this.schema.datasource.filter)
       : Object.assign({}, this.schema.datasource.filter)
 
+    // Allow for api config alias
+    if (schema.datasource.source && schema.datasource.source.api) {
+      if (!config.get('api')[schema.datasource.source.api]) {
+        callback(
+          new Error(
+            `Settings for API '${
+              schema.datasource.source.api
+            }' not found in configuration file!`
+          )
+        )
+      } else {
+        this.source = Object.assign(
+          {},
+          schema.datasource.source,
+          config.get('api')[schema.datasource.source.api]
+        )
+      }
+    } else {
+      this.source = schema.datasource.source
+    }
+
+    // Default to dadiapi
     if (!this.source.type) {
       this.source.type = 'dadiapi'
     }
 
+    // Default options if not provided for dadiapi
+    if (!this.source.host && this.source.type === 'dadiapi') {
+      let apiInfo = {}
+
+      // If there is only one config in the api block
+      if (
+        config.get('api').host ||
+        config.get('api').port ||
+        config.get('api').auth
+      ) {
+        apiInfo = config.get('api')
+      } else {
+        // Else, it's probably an object of configs so use first 'type=dadiapi' creds we see or the first with no type defined
+        Object.keys(config.get('api')).map(i => {
```

Comment:
You mention in the comment that you're getting the *first* element with type `dadiapi` or no type, but I believe you're actually getting the last one. You'll cycle through all the keys with `map` and rewrite `apiInfo` every time there's a match, no?

If you want just the first match, perhaps you could use a `find` instead of a `map`, which also saves you from having to cycle through all the keys if you find a match early on?

Reply:
Good point!

Reply:
have tweaked, which changes the order slightly:

```If no `api` is specified the fallback is:
* The "original way" above
* Or an `api` block named explicitly as `"dadiapi": { ... } `
* Or first `api` defined with `"type": "dadiapi"`
* Or first `api` with no defined `type`
* Or settings in the `source` of the datasource itself (which can override if matched above)
```